server/src/test/java/org/elasticsearch/action/admin/indices/datastream/DeleteDataStreamRequestTests.java

```diff
@@ -85,6 +90,31 @@ public void testDeleteDataStream() {
         }
     }
 
+    public void testDeleteSnapshottingDataStream() {
```

Comment:
I think this would benefit from a real integration test like we have for indices with `org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT#testCloseOrDeleteIndexDuringSnapshot`. There's lots of things that can go wrong with changes to the snapshot code that this test won't catch. I think it's pretty easy to create an analogous test to `org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT#testCloseOrDeleteIndexDuringSnapshot` for data streams right?

Reply:
You're right, I've added `SharedClusterSnapshotRestoreIT.testDeleteDataStreamDuringSnapshot`